File: Golang Solutions/Find the Town Judge.go
func findJudge(N int, trust [][]int) int {
    if N==1{
        return 1
    }
    m:= make(map[int]map[int]bool)
    j,cnt:=0,0
    for _,x:= range trust{
        inner,ok:=m[x[0]]
        if(!ok){
            inner=make(map[int]bool)
            m[x[0]]=inner
        }
        inner[x[1]]=true
        
    }
    for i:=1;i<=N;i++{
        _,ok:=m[i]
        if(!ok){
             j=i
            cnt++
        }
    }
    if j==0 && cnt>1{
        return -1
    }
    for i:=1;i<=N;i++{
        _,ok:=m[i]
        if(ok){
            if m[i][j]==false{
                return -1
            }
        }
    }
    return j
        
    
}

File: Find the Town Judge.cpp
class Solution {
public:
    int findJudge(int N, vector<vector<int>>& trust) {
        if(N==1)
            return 1;
        
        bitset<1000> x[N+1];
        int j,cnt=0;
        bitset<1000>z;
        z.set();
        for(int i=0;i<trust.size();i++)
        {
            x[trust[i][0]][trust[i][1]]=1;      
        }
        
        for(int i=1;i<=N;i++)
        {
            
            if(!x[i].any())
            {
                j=i;
                cnt++;
            }  
            else
            {
                z = z & x[i];
            }
        }
        if(j==0 || cnt >1)
            return -1;
        if(j== z._Find_first() )
            return j;
        else
            return -1;
               
        
        
    }
};
